src/Network/X509/SelfSigned.hs

```diff
@@ -655,7 +656,13 @@ unsafeMakeCredential (X509CertChainPem cert chain) (X509KeyPem keyBytes) =
 --
 data TlsPolicy
     = TlsInsecure
-    | TlsSecure Bool (ServiceID -> IO (Maybe Fingerprint))
+    | TlsSecure
+        { _tlsPolicyUseSystemStore :: !Bool
+            -- ^ whether to use the system certificate store.
+        , _tlsPolicyCallback :: !(ServiceID -> IO (Maybe Fingerprint))
+            -- ^ a callback for looking up certificate fingerprints for known
+            -- service endpoints.
+        }
```

Comment:
Does this not throw a "partial fields" warning?

Reply:
There haven't been any warnings in the nix build